### PR TITLE
Adding description and links to course archive

### LINKED ISSUE #1114: Add description and anchor links to course archive page
It would be great if the top of [the course archive page](https://learn.wordpress.org/courses/) included a quick description/intro of what the page is all about (this must still be written - I'll circle back to this issue with the text) as well as anchor links to each of the categories on the page. This will make page navigation a bit easier, as well as prevent the issue of people only seeing development courses above the fold and not scrolling down (which is an issue that has been reported anecdotally).

Here is an imperfect mockup I made in the inspector to show what I mean:
Screenshot 2022-11-23 at 11 12 40 AM

### PR DESCRIPTION
**Pull request by hlashbrooke**
This PR adds a page description and anchor links to the course archive template. Fixes #1114.

### CODE REVIEW

**Comment by hlashbrooke** [comment]
This is complete:

Screenshot 2022-11-29 at 2 04 14 PM

The anchor links are dynamic and come from the same array of data as the courses below, so they will be in the same order as the categories on the page.

File: wp-content/themes/pub/wporg-learn-2020/archive-course.php [hunk 1]
```diff
@@ -31,40 +32,85 @@
...
-						// Close the previous card-grid if there was a previous category
-						if ( ! empty( $prev_category ) ) {
-							echo '</div>';
-						}
-						?>
-		<h2 class="h4 course-category-header"><?php echo esc_html( $category_title ); ?></h2>
+				endwhile; ?>
+
+				<?php
+				// $courses_list .= '</div>';
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2020/archive-course.php]: Was it intentional to leave this?

Comment (hlashbrooke) [reply wp-content/themes/pub/wporg-learn-2020/archive-course.php]: Oh good catch - will remove now.

File: wp-content/themes/pub/wporg-learn-2020/css/components/_section.scss [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+		*:last-child {
+			margin-bottom: 0;
+		}
+	}
+
+	&--with-space {
+		padding-bottom: 28px;
+	}
+}
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2020/css/components/_section.scss:45]: Was `_section-heading.scss`

File: wp-content/themes/pub/wporg-learn-2020/archive-course.php [hunk 3]
```diff
@@ -9,62 +9,101 @@
...
 get_header();
 get_template_part( 'template-parts/component', 'breadcrumbs' );
 ?>
 
 <main id="main" class="site-main">
-
 	<section>
-		<div class="row align-middle between section-heading section-heading--with-space gutters">
-			<?php the_archive_title( '<h1 class="section-heading_title h2 col-8">', '</h1>' ); ?>
-			<?php get_template_part( 'template-parts/component', 'archive-search' ); ?>
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2020/archive-course.php:25]: Moved to `intro` section, matching Lesson Plan landing page layout

| Lesson Plans | Courses |
|-|-|
| Screen Shot 2022-11-30 at 1 38 46 PM | Screen Shot 2022-11-30 at 1 30 08 PM |

File: wp-content/themes/pub/wporg-learn-2020/archive-course.php [hunk 4]
```diff
@@ -9,62 +9,101 @@
...
+				</div>
+				<nav class="section-nav">
+					<ul>
+					<?php foreach ( $all_courses as $slug => $category ) { ?>
+						<li class="section-nav-item">
+							<a href="#<?php echo esc_attr( $slug ); ?>"><?php echo esc_html( $category['name'] ); ?></a>
+						</li>
+					<?php } ?>
+					</ul>
+				</nav>
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2020/archive-course.php:69]: Standard markup for an internal navigation, see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav

Using semantic HTML for these helps assistive technologies.

**Review by adamwoodnz — APPROVED** [review]
LGTM :shipit:

**Pull request merged**